### ISSUE TITLE
Events on initial load 

### ISSUE THREAD

**Issue by beardyw** [json-editor/json-editor#501]
I can't see any other reference to this. On initial load of the editor core.js fires events as 

        self.trigger('ready');
        self.trigger('change');

If you are listening for user change events it seems reasonable to wait until after "ready" is fired. However in doing so you still then get a change event from the initial load. It might make sense to reorder these so that "ready" comes last. 

It works for me. Happy to make the change unless there is some deep reason for this order.

**Comment by pmk65** [comment]
@beardyw The initial change event is pretty useless as you explained, unless you can catch changes made by setting the "startval" property.
I haven't tested this, so I don't know if it's possible or if that triggers another "change" event. If it does triggers a 2nd "change" event, I see no reason why we shouldn't change the order.

**Comment by pmk65** [comment]
@beardyw I did some testing, and the initial change event contains the values set by `startval`, so I don't think it's a good idea to change it.

Instead `bind` a counter to the change event. Then if you want to skip the initial change event, check if the value of the counter>1. Example: https://is.gd/zEQvXy
````javascript
var jseditor = new window.JSONEditor(document.querySelector("#json-editor-form"), jedata);

var changeFunc = function() {
  console.log('Change event triggered', JSON.stringify(jseditor.getValue()));
  console.log('Change counter', this.getCounter());
}.bind({
  counter: 0,
  getCounter: function() { return ++this.counter;}
});

var readyFunc = function() {
  console.log('Ready event triggered');
  jseditor.on('change', changeFunc);
};

jseditor.on('ready', readyFunc);
````

**Comment by beardyw** [comment]
@pmk65 Thanks for looking at this.  Based on your suggestion I have realised that others may have coded it that way, and the change I suggest would then break their code. 

All the same, I am not really convinced that something we call startval should be regarded as a change after the ready event. I wasn't proposing to remove it, just place it where it makes more sense - as a (kind of) change before we are ready. However it is probably best left as it is as you say. Thanks

**Comment by pmk65** [comment]
@beardyw I played around with it a bit and found a really good reason to keep it as it is. And why startval should be included in the onchange event.

Take a look at this example: https://is.gd/v10mAd
It uses the counter method I described before, to store the initial value of the form (counter==1)
Then on the subsequent onchange events, it compares the current value to the previous value and returns (console.log) the difference.
